ssz/tree_hash/tree_hash.py

```diff
@@ -30,19 +32,38 @@
 
 
 def hash_tree_root(input_object: Any, sedes: Any=None) -> Hash32:
+    result = _hash_tree_root(input_object, sedes)
+    if len(result) < 32:
+        return Hash32(result.ljust(32, b'\x00'))
+    return Hash32(result)
+
+
+def _hash_tree_root(input_object: Any, sedes: Any=None) -> bytes:
     if sedes is None:
         sedes = infer_sedes(input_object)
 
-    if isinstance(sedes, (Boolean, Hash, UnsignedInteger)):
-        serialization = sedes.serialize(input_object)
-        if isinstance(sedes, UnsignedInteger) and sedes.num_bytes > 32:
-            return hash_eth2(serialization)
+    if isinstance(sedes, (Boolean, Hash)):
```

Comment:
I would feel a little bit safer if each `if` would be an `elif` (with an `else` for the exception at the end). This way, if one forgets a return statement, only an exception is raised instead of a wrong value being returned.